src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -134,6 +113,26 @@ export default class GooglePayPaymentProcessor {
                 ]).then(([googlePay, paymentDataRequest]) => {
                     this._googlePayClient = this._getGooglePayClient(googlePay, testMode);
                     this._paymentDataRequest = paymentDataRequest;
+
+                    return this._googlePayClient.isReadyToPay({
+                        allowedPaymentMethods: [
+                            {
+                                type: paymentDataRequest.allowedPaymentMethods[0].type,
+                                parameters: {
+                                    allowedAuthMethods: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedAuthMethods,
+                                    allowedCardNetworks: paymentDataRequest.allowedPaymentMethods[0].parameters.allowedCardNetworks,
+                                },
+                            },
+                        ],
+                        apiVersion: paymentDataRequest.apiVersion,
+                        apiVersionMinor: paymentDataRequest.apiVersionMinor,
+                    }).then(response => {
+                        if (response.result) {
+                            Promise.resolve();
+                        }
+
+                        Promise.reject();
```

Comment:
You know you haven't returned anything here. So the promise will resolve successfully even if `response.result` is `true`. You need to write a `return` statement or simply do a `throw`. Preferably you should also provide a rejection reason. For example:

```ts
return Promise.reject(response); // Or alternatively you can throw a custom error, i.e.: throw new PaymentMethodInvalidError();
```